### PR TITLE
Add support for composable index templates

### LINKED ISSUE #404: [PROPOSAL] - Add missing high-level client methods for handling API Guide examples.
### What are you proposing?
Currently the some of the API methods related to composable templates do not yet exist in the high-level client, they only exist as methods on the low-level client. This proposal seeks to add the missing methods to the high-level client so that the low-level client examples in the `/guides/index_template.md` can be replaced with the high-level client implementations.

### What problems are you trying to solve?
By adding the missing methods to the high-level client, this would allow for updating the API action guides to use the more C# friendly implementation via the high-level client.

#OSCI-2023

### PR DESCRIPTION
**Pull request by Xtansia**
### Description
Adds support for composable index templates

### Issues Resolved
Closes #404 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Xtansia** [comment]
@dblock When you have time could you review this please?

File: .github/workflows/integration.yml [hunk 1]
```diff
@@ -62,12 +62,13 @@ jobs:
           path: client/build/output/*
 
   integration-opensearch-unreleased:
+    if: false # TODO: Temporarily disabled due to failures building & running OpenSearch from source, pending investigation & fixes
```
Comment (dblock) [inline .github/workflows/integration.yml:65]: Open an issue and link it from here.

File: guides/index-template.md [hunk 2]
```diff
@@ -19,186 +17,169 @@ var config = new ConnectionSettings(node)
...
 ```
 
 When we create an index named `books-fiction-horror`, OpenSearch will apply the `books-fiction-*` template's settings, and `books_mappings` template mappings to the index:
 
 ```csharp
-client.Indices.Create("books-fiction-horror");
-var getResponse = client.Indices.Get("books-fiction-horror");
-Console.WriteLine(getResponse.Indices["books-fiction-horror"].Settings.NumberOfShards); // 1 Console.WriteLine(getResponse.Indices["books-fiction-horror"].Mappings.Properties["pages"].Type); // integer 
+createIndex = await client.Indices.CreateAsync("books-fiction-horror");
+Debug.Assert(createIndex.IsValid, createIndex.DebugInformation);
```
Comment (dblock) [inline guides/index-template.md:143]: I think for a guide we may want to `Console.WriteLine` instead of `Debug.Assert`.

**Review by dblock — APPROVED** [review]
Did a cursory review, looks ok to me.

**Comment by Xtansia** [comment]
@dblock Have made your suggested changes

**Review by dblock — APPROVED** [review]

**Pull request merged**